### ISSUE TITLE
Joint2D bug

### ISSUE THREAD

**Issue by tatarinovvladimir** [godotengine/godot#46571]
**Godot version:**
3.2.4 RC3 official

**OS/device including version:**
Windows 10 

**Issue description:**
Joint2D removes one of the connected nodes after saving and restarting the scene.

**Steps to reproduce:**

1. Create some RigidBody2d and join it by multiple Joint2D.
2. Save and restart scene.
3. Close the scene.
4. Run project
5. Open scene with Joint2D again.

After repeating these actions when opening the scene, some Joint2D lose one of the two selected nodes (A or B)

**Comment by pouleyKetchoupp** [comment]
I'm not able to reproduce this issue in 3.2.4 RC3. Could you please provide a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to help with investigating?

**Comment by tatarinovvladimir** [comment]
> I'm not able to reproduce this issue in 3.2.4 RC3. Could you please provide a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to help with investigating?

Strange, but I also can't reproduce this in a new project. Apparently this is a problem due to the transfer of the project from 3.2.3 to 3.2.4. It seems that to solve this, you need to recreate the Join2D node in the old project. Since the creation of the 3.2.1 project, all transitions to a newer version did not require any changes.

**Comment by pouleyKetchoupp** [comment]
Yeah I agree that even if it's just in a 3.2.3 project, it's a regression that needs fixing. I'm still not able to reproduce this case by making a 3.2.3 project though. Would you be able to share your project or a piece of it to help reproduce the issue?

**Comment by tatarinovvladimir** [comment]
> Yeah I agree that even if it's just in a 3.2.3 project, it's a regression that needs fixing. I'm still not able to reproduce this case by making a 3.2.3 project though. Would you be able to share your project or a piece of it to help reproduce the issue?


So I cut and packed a piece of the project, here it will definitely be reproduced.
What do we have to do:

First start Godot 3.2.3

1. Open the scene Resources - Cars - Ranniy - Car
2. Open the scene Resources - Cars - Barzha - Car
3. Run them as a separate scene, look at the work of the suspension.
4. Save and close the scenes and go to step 1 again.
5. No matter how many times you repeat, everything should work.

Next start Godot 3.2.4

1. Unlock the scene Resources - Cars - Ranniy - Car
2. Open the scene Resources - Cars - Barzha - Car
3. Run them as a separate scene, look at the work of the suspension.
4. Save and close the scenes and go to step 1 again.
5. After several repetitions, the suspension will simply collapse due to a bug in Join2D
[test.zip](https://github.com/godotengine/godot/files/6065161/test.zip)

**Comment by pouleyKetchoupp** [comment]
Thanks! I can confirm it's a regression from #44704 (cc @madmiraal).

I was able to reproduce the issue and I've made a simpler repro case now:
[joint-create-test.zip](https://github.com/godotengine/godot/files/6065321/joint-create-test.zip)

Steps in this project with 2.3.4 RC3:
- Open Node2D-A.tscn
- Open Node2D-B.tscn
- Switch back to Node2D-A.tscn
- Observe warning on joint, one of the bodies has been reset.
